### PR TITLE
add SLES support.

### PR DESCRIPTION
**Pull request by tuxmea**
Sles does not have packages for some depending software.
we use archive in this case

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -216,14 +216,16 @@
 
   Hash $grafana_config,
 
+  Optional[String[1]] $archive_source = undef,
```
Comment (genebean) [inline manifests/init.pp:219]: Do we want to allow any string or just URLs like so?

```suggestion
  Optional[Stdlib::HTTPUrl] $archive_source = undef,
```

File: manifests/repos.pp [hunk 2]
```diff
@@ -54,6 +54,10 @@
         }
       }
 
+      'Suse': {
+        # Suse nutzt archive, kein repo
+      }
```
Comment (genebean) [inline manifests/repos.pp:59]: Is this manifest actually included when repo management is disabled? I would think that on SUSE this should fail since we are not managing the repo.

File: manifests/service.pp [hunk 3]
```diff
@@ -4,31 +4,49 @@
...
+        source => '/opt/influxdb/usr/lib/influxdb/scripts/influxdb.service',
+        notify => Exec['/usr/bin/systemctl daemon-reload'],
+      }
+      exec { '/usr/bin/systemctl daemon-reload':
+        refreshonly => true,
+      }
+      service { $puppet_metrics_dashboard::influx_db_service_name:
+        ensure => running,
+        enable => true,
+      }
```
Comment (genebean) [inline manifests/service.pp:20]: I would think we should defer management of this to [camptocamp/systemd](https://forge.puppet.com/camptocamp/systemd)

File: spec/classes/init_spec.rb [hunk 4]
```diff
@@ -35,6 +35,12 @@
         # verify the defualts have not changed
         # rubocop:disable RSpec/ExampleWording
         it 'should have all the expected default vaulues for parameters' do
+          case facts[:osfamily]
```
Comment (genebean) [inline spec/classes/init_spec.rb]: Please use the non-legacy version of this

File: spec/classes/install_spec.rb [hunk 5]
```diff
@@ -18,45 +18,51 @@
           PRE_COND
         end
 
-        it do
-          is_expected.to contain_package('influxdb')
-            .with_ensure('present')
-        end
+        case facts[:osfamily]
```
Comment (genebean) [inline spec/classes/install_spec.rb:21]: Please use the non-legacy version of this

File: spec/classes/install_spec.rb [hunk 6]
```diff
@@ -18,45 +18,51 @@
...
-      context 'with Kapacitor and Chronograf enabled' do
-        let(:pre_condition) do
-          <<-PRE_COND
-            class {'puppet_metrics_dashboard':
-              enable_kapacitor  => true,
-              enable_chronograf => true,
-            }
-          PRE_COND
-        end
+      case facts[:osfamily]
```
Comment (genebean) [inline spec/classes/install_spec.rb:40]: Please use the non-legacy version of this

File: spec/classes/post_start_configs_spec.rb [hunk 7]
```diff
@@ -2,6 +2,12 @@
 
 describe 'puppet_metrics_dashboard::post_start_configs' do
   on_supported_os.each do |os, facts|
+    case facts[:osfamily]
```
Comment (genebean) [inline spec/classes/post_start_configs_spec.rb]: Please use the non-legacy version of this

File: manifests/init.pp (hunk 1 above)
Comment (tuxmea) [reply manifests/init.pp:219]: It can also be a local share or any http/ftp url.

File: manifests/repos.pp (hunk 2 above)
Comment (tuxmea) [reply manifests/repos.pp:59]: I am going to disable repo management on sles using hiera data and remove the sles lines from puppet code.

File: manifests/service.pp [hunk 8]
```diff
@@ -4,31 +4,44 @@
...
-      ],
+  case $facts['os']['family'] {
+    'Suse': {
+      systemd::unit_file { "${puppet_metrics_dashboard::influx_db_service_name}.service":
+        source => "puppet:///modules/${module_name}/influxdb.service",
+      }
+      ~> service { $puppet_metrics_dashboard::influx_db_service_name:
+        ensure => running,
+        enable => true,
+      }
```
Comment (genebean) [inline manifests/service.pp:15]: I am not going to block on this but would appreciate if you'd use something like this instead of the chaining arrows - I personally find them hard to follow and easy to overlook.

```suggestion
      service { $puppet_metrics_dashboard::influx_db_service_name:
        ensure    => running,
        enable    => true,
        require   => Systemd::Unit_file["${puppet_metrics_dashboard::influx_db_service_name}.service"],
        subscribe => Systemd::Unit_file["${puppet_metrics_dashboard::influx_db_service_name}.service"],
      }
```

Comment (tuxmea) [reply manifests/service.pp:15]: I always thought that subscribe automatically does the require?

Comment (genebean) [reply manifests/service.pp:15]: I honestly don't know if it does or doesn't ...

Comment (tuxmea) [reply manifests/service.pp:15]: It does (according to code documentation) https://github.com/puppetlabs/puppet/blob/main/lib/puppet/type.rb#L1652

Comment (genebean) [reply manifests/service.pp:15]: Awesome, thanks for educating me! (sincerely)

Comment (tuxmea) [reply manifests/service.pp:15]: more precise: https://github.com/puppetlabs/puppet/blob/main/lib/puppet/type.rb#L1646 The direction in is the ordering used in to_edges: https://github.com/puppetlabs/puppet/blob/main/lib/puppet/type.rb#L1580

**Comment by tuxmea** [comment]
I have tested a complete fresh installation on sles12. working without issues.

**Comment by tuxmea** [comment]
All change requests are implemented. Do we need a Jira ticket for this PR?

File: data/os/Suse.yaml [hunk 9]
```diff
@@ -0,0 +1,5 @@
+---
+puppet_metrics_dashboard::manage_repos: false
+puppet_metrics_dashboard::influx_db_service_name: 'influxdb'
+puppet_metrics_dashboard::grafana_version: '7.1.4'
```
Comment (jarretlavallee) [inline data/os/Suse.yaml:4]: Does 7.1.4 see the same issues as described in #45?

**Review by MartyEwings — COMMENTED** [review]
If we resolve the conflicts and force push the addition of the SLES OS to the metadata should trigger full acceptance test run on the SLES OS and allow us to merge this

**Comment by MartyEwings** [comment]
Close to re-open to trigger test run

**Comment by MartyEwings** [comment]
superseded by #175
